### DISCUSSION TITLE
Request Parameters for Feeds not respecting after parameter

### DISCUSSION THREAD

**Discussion by rpavery** [transitland/transitland#431] [Q&A] [answered]
Hello -

I'm trying to limit the results I get from listing all the feeds for an agency (based on the Onestop feed id) to just those after a certain feed number. The documentation for the [feeds API](https://www.transit.land/documentation/rest-api/feeds) notes that the `after` parameter will 'Return records past this record ID'. However, when I try doing so, it does not work as expected.

Examples: (all with my API key removed) Each of these will give the full list of results for the after id provided, but incrementing the after ID by 1 will result in empty feed lists for all of the agencies.

- [King County Metro](https://transit.land/api/v2/rest/feeds/f-c23-metrokingcounty?api_key=<api_key>&after=27)
- [Community Transit](https://transit.land/api/v2/rest/feeds/f-c29-communitytransit?api_key=<api_key>&after=335)
- [Sound Transit](https://transit.land/api/v2/rest/feeds/f-c23-soundtransit?api_key=<api_key>&after=338)
- [Seattle Monorail](https://transit.land/api/v2/rest/feeds/f-seattlemonorail~wa~us?api_key=<api_key>&after=2289)

There is no apparent rhyme or reason to the numbers where the after link stops working. I was expecting to, for example, limit the King County Metro list of feeds to only those feeds AFTER feed id 369461. Can somebody please let me know, is this feature broken, or does it not work the way it is described on the API page.

Note - I am NOT using this for pagination - I'm simply limiting the feed list based on the after parameter described in the Feed API.

Thanks!
Ryan

**Comment by irees** [comment, marked as answer]
hi @rpavery 

The `after` parameter should be treated as an opaque value for pagination with no other purpose. It's an integer value that happens to match the integer id for a particular feed (for reasons of historical compatibility with some api clients), but this may not always be true in the future (e.g. switch to some kind of encoded string that makes it clear it's a server controlled value.)

Requesting a particular feed with the path parameter, e.g. `https://transit.land/api/v2/rest/feeds/f-c23-metrokingcounty` is the same as setting the `feed_key` or `onestop_id` parameter, depending on the specific value. The result set will be a single feed with the requested key, and no other feeds. In this case, incrementing the `after` parameter advances past that feed and thus returns an empty result set.

We will update the documentation to clarify that `after` cannot be used with one of the path or key based parameters, and possibly change the server behavior to return an error.
